src/hock/ElementQueryHock.jsx

```diff
@@ -28,58 +28,56 @@ if(typeof window !== 'undefined') { // Don't try to detect resize events on serv
- *                                                        widths allowed by this element query. If
- *                                                        the second array item is excluded it is
- *                                                        assumed that there is no maximum.
- * @param {Number[]} [eqs[].heightBounds=[0, Infinity]] - An array containing two values for the
- *                                                        minimum (inclusive) and maximum (exclusive)
- *                                                        heights allowed by this element query. If
- *                                                        the second array item is excluded it is
- *                                                        assumed that there is no maximum.
- * @return {HockApplier}
+ * @param {Array<ElementQueryObject>} eqs An array of element queries to check for
+ * @return {ElementQueryApplier}
  */
 
-const ElementQueryHock = (eqs: ElementQuery[]): HockApplier => {
-    return (ComposedComponent: ReactClass<any>): ReactClass<ElementQueryHock> => {
-        return class ElementQueryHock extends Component {
+const ElementQueryDecorator = (eqs: ElementQuery[]): HockApplier => {
+    return (ComposedComponent: ReactClass<any>): ReactClass<any> => {
+
+        /**
+         * @component
+         *
+         * ElementQueryHock is designed to allow components to respond to the size of their containing
+         * element rather than to the size of the window. When used on a component, your component will receive some extra props.
+         *
+         * @example
+         * const example = (props) => {
+         *     if(!props.eqReady) return <div>No data yet</div>;
+         *     return <div>
+         *         <div>width: {props.eqWidth}</div>
+         *         <div>height: {props.eqHeight}</div>
+         *         <div>active queries: {props.eqActive.join(', ')}</div>
+         *         <div>inactive queries: {props.eqInactive.join(', ')}</div>
+         *     </div>
+         * }
+         *
+         * const ElementQueryHockExample = ElementQueryHock([
+         *     {
+         *         name: 'medium',
+         *         widthBounds: [300, 600],
+         *         heightBounds: [200, 400]
+         *     },
+         *     {
+         *         name: 'large',
+         *         widthBounds: [600, 1200],
+         *         heightBounds: [400, 1800]
+         *     }
+         * ])(example);
+         *
+         * @childprop {number} eqWidth The current width of the parent element.
+         * @childprop {number} eqHeight The current height of the parent element.
+         * @childprop {Array<string>} eqActive An array of currently active queries.
+         * @childprop {Array<string>} eqInactive An array of currently inactive queries.
+         * @childprop {boolean} eqReady An boolean that can be used to determine whether the ElementQueryHock has been able.
```

Comment:
Would be good to update 'An boolean' to 'A boolean' too, thought that one's my bad